target/linux/ramips/dts/mt7621_sztozed_zlt-s12-pro.dts

```diff
@@ -0,0 +1,256 @@
+			gpios = <&gpio 24 GPIO_ACTIVE_LOW>;
+		};
+
+		signal3 {
+			label = "blue:signal3";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+		};
+
+		signal4 {
+			label = "blue:signal4";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+		};
+
+		signal5 {
+			label = "blue:signal5";
+			gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
+		};
+
+		phone {
+			label = "blue:phone";
+			gpios = <&gpio 31 GPIO_ACTIVE_LOW>;
+		};
+
+		slic_clk_switch {
+			label = "slic_clk_switch";
+			gpios = <&gpio 32 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	gpio_export {
+		compatible = "gpio-export";
+
+		lt72_ant0 {
+			gpio-export,name = "lt72_ant0";
+			gpio-export,input = <0>;
+			gpios = <&gpio 6 GPIO_ACTIVE_HIGH>;
+		};
+
+		lt72_ant1 {
+			gpio-export,name = "lt72_ant1";
+			gpio-export,input = <0>;
+			gpios = <&gpio 7 GPIO_ACTIVE_HIGH>;
+		};
+
+		lt72_power {
+			gpio-export,name = "lt72_power";
+			gpio-export,output = <1>;
+			gpios = <&gpio 14 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	spidev@1 {
+		spi-cpol = <0x01>;
+		compatible = "rohm,dh2228fv";
+		spi-cpha = <0x01>;
+		linux,modalias = "spidev";
+		status = "okay";
+		reg = <0x01>;
+		spi-max-frequency = <0x1f4000>;
+	};
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x8000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@38000 {
+				label = "tozed-conf";
+				reg = <0x38000 0x8000>;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0xfb0000>;
+			};
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	mt76@0,0 {
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
+
+&pcie1 {
+	mt76@0,0 {
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+		ieee80211-freq-limit = <2400000 2500000>;
+	};
+};
+
+&ethernet {
+	pinctrl-0 = <&mdio_pins>, <&rgmii1_pins>;
+};
+
+&gmac0 {
+	nvmem-cells = <&macaddr_factory_4>;
+	nvmem-cell-names = "mac-address";
+};
+
+&switch0 {
+	ports {
+		port@0 {
+			status = "okay";
+			label = "lan4";
+		};
+
+		port@1 {
+			status = "okay";
+			label = "lan3";
+		};
+
+		port@2 {
+			status = "okay";
+			label = "lan2";
+		};
+
+		port@3 {
+			status = "okay";
+			label = "lan1";
+		};
+	};
+};
+
+&state_default {
```

Comment:
@arinc9 Your PR removes the gpio section. What about this pcm one?

Reply:
The pin group and its function must be defined under the pinctrl node.
uart2 pin group is already defined on mt7621.dtsi so we can just do this:
```
&uart2_pins {
	uart2 {
		function = "pcm";
	};
};
```
This DT is similar:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/dts/mt7621_mediatek_ap-mt7621a-v60.dts

Then, the pin group must be claimed under its consumer node. Which would be the pcm node, but I don't see it on mt7621.dtsi. I doubt there's even a driver for it on upstream or OpenWrt. There's only the i2s driver added by this patch:
https://github.com/openwrt/openwrt/blob/617fcd4879a7c71810648a2c398e0057ee1073bb/target/linux/ramips/patches-5.10/835-asoc-add-mt7620-support.patch#L862

For example, mt7628an.dtsi has a pcm node.
https://github.com/openwrt/openwrt/blob/ad6b0770493e0fd332db9a7fe09f9214342c578b/target/linux/ramips/dts/mt7628an.dtsi#L230

Reply:
Would @1Conan explain why the pcm interface needs to be used on this particular device?